server/py/framework/utils/clients/messaging.py

```diff
@@ -312,3 +411,10 @@ def _prepare_request_data(self, method: str, path: str):
         path = path.removeprefix(f"{prefix}/").removeprefix(f"{version}/")
         service_instance = self._discovery.resolve_service_by_request(method, path)
         return path, version, service_instance
+
+    @staticmethod
+    def _resolve_full_request_path(path, service_instance, version):
+        # The service and version prefixes have been removed from the path earlier in the process.
```

Comment:
This happens in `_prepare_request_data`, which is currently only called in `proxy_request_to_service`. 
Will the new `delete` method also be called via `proxy_request_to_service`? I don't think so.

Reply:
No, `_prepare_request_data` is not used in sync communication because it assumes the path contains the service name and api version but I wanted to keep the sync client agnostic